internal/component/prometheus/remote/queue/filequeue/filequeue.go

```diff
@@ -0,0 +1,182 @@
+package filequeue
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+	"github.com/vladopajic/go-actor/actor"
+)
+
+var _ actor.Worker = (*queue)(nil)
+var _ types.FileStorage = (*queue)(nil)
+
+// queue represents the on disk queue. This is an ordered list by id in the format: <id>.committed
+// Each file contains a byte buffer and an optional metatdata map.
+type queue struct {
+	self      actor.Actor
+	directory string
+	maxIndex  int
+	logger    log.Logger
+	files     actor.Mailbox[types.Data]
```

Comment:
I'd prefer to call this `inbox`? It's a mailbox with Data, not files yet.

Reply:
Inbox feels a little generic, but agree files is wrong.